src/buildercore/core.py

```diff
@@ -287,6 +287,9 @@ def stack_all_ec2_nodes(stackname, workfn, username=config.DEPLOY_USER, concurre
     if node:
         nodes = {k: v for k, v in nodes.items() if v == int(node)}
         public_ips = {k: v for k, v in public_ips.items() if k in nodes.keys()}
+    elif instance_ids:
+        nodes = {k: v for k, v in nodes.items() if k in instance_ids}
+        public_ips = {k: v for k, v in public_ips.items() if k in nodes.keys()}
```

Comment:
```python
    if node:
        nodes = {k: v for k, v in nodes.items() if v == int(node)}
    elif instance_ids:
        nodes = {k: v for k, v in nodes.items() if k in instance_ids}
    public_ips = {k: v for k, v in public_ips.items() if k in nodes.keys()}
```

Reply:
Not sure if equivalent, because `public_ips` is not present for the missing `else` case but it's executed in the second version.

Reply:
you're right, ignore me